### ISSUE TITLE
p-toggleButton: not accessible anymore

### ISSUE THREAD

**Issue by MaSch0212** [primefaces/primeng#18250]
### Describe the bug

After issue #17901 the toggle button is not accessible anymore. It simply shows as StaticText in Chromes Accessibility view. Previously the p-toggleButton component included a native button element that handled accessibility. This is now gone.

I noticed this issue only because we are using the p-selectButton, which internally uses the p-toggleButton. With this change you are not able to identify the selected element in that select button if you are blind. (or in our case our playwright tests fail ^^)

### Reason for not contributing a PR

- [ ] Lack of time
- [ ] Unsure how to implement the fix/feature
- [ ] Difficulty understanding the codebase
- [x] Other

### Other Reason

Not sure why the issue #17901 has been implemented. I do not want to just rollback the change. Some clarification is necessary.

### Reproducer

https://primeng.org/togglebutton

### Environment

Standard Angular 19.2.2 app with PrimeNG 19.1.2.

### Angular version

19.2.2

### PrimeNG version

v19

### Node version

19.1.2

### Browser(s)

Chrome

### Steps to reproduce the behavior

1. Go to https://primeng.org/togglebutton
2. Open the Chrome DevTools and enable accessibility tree
   ![Image](https://github.com/user-attachments/assets/c3728574-c9ce-4a5d-9ac8-2dadf2170762)
3. Find one of the toggle buttons and inspect it with accessibility tree turned on
   ![Image](https://github.com/user-attachments/assets/c547ffff-d4a8-4b80-8b90-d48bcb3f2b9e)

### Expected behavior

I am not 100% sure which aria-role is the best. But I guess role "button" with the "aria-checked" or "aria-pressed" set to whatever the state is.

Edit: I checked again in our Playwright tests. In 19.0.10 we searched for role "button" and checked for "aria-pressed" attribute.

**Comment by Will-Smith-007** [comment]
This issue was resolved with f9e46cd75c53e6de07f1e03440fc6a2d25786f6b and can be closed.
